pkg/apis/apiresource/v1alpha1/apiresourceimport_helpers.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2021 The Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetCondition sets the status condition. It either overwrites the existing one or creates a new one.
+func (apiResourceImport *APIResourceImport) SetCondition(newCondition APIResourceImportCondition) {
```

Comment:
style nit throughout: `apiResourceImport` in the type receiver is a bit long and doesn't add much. Could it just be `func (i *APIResourceImport) SetCondition(...)` ?